src/api/v1/community_businesses/volunteers/get.ts

```diff
@@ -0,0 +1,135 @@
+import * as Hapi from 'hapi';
+import * as Boom from 'boom';
+import { has, pick, omit } from 'ramda';
+import { User, ModelQuery, Volunteers } from '../../../../models';
+import { query, filterQuery, response } from '../../users/schema';
+import { id } from '../schema';
+import { getCommunityBusiness, isChildOrganisation } from '../../prerequisites';
+import { GetAllVolunteersRequest } from '../../types';
+
+
+const routes: Hapi.ServerRoute[] = [
+  {
+    method: 'GET',
+    path: '/community-businesses/{organisationId}/volunteers',
+    options: {
+      description: 'Retreive list of all volunteers from an organisation',
+      auth: {
+        strategy: 'standard',
+        access: {
+          scope: ['user_details-child:read'],
+        },
+      },
+      validate: {
+        params: { organisationId: id },
+        query: {
+          ...query,
+        },
+        failAction: (request, h, err) => err,
+      },
+      response: { schema: response },
+      pre: [
+        { method: getCommunityBusiness, assign: 'communityBusiness' },
+        { method: isChildOrganisation, assign: 'isChild' },
+      ],
+    },
+    handler: async (request: GetAllVolunteersRequest, h: Hapi.ResponseToolkit) => {
+      const { query, pre: { communityBusiness, isChild }, server: { app: { knex } } } = request;
+
+      if (request.params.organisationId !== 'me' && !isChild) {
+        return Boom.forbidden('Insufficient permissions to access this resource');
+      }
+
+      const q: {
+        limit?: number,
+        offset?: number,
+        order?: [string, 'asc' | 'desc']
+      } = {
+        ...pick(['limit', 'offset'], query),
+        order: query.sort ? [query.sort, query.order || 'asc'] : undefined,
+      };
+
+      const modelQuery: ModelQuery<User> = { ...q, where: { deletedAt: null } };
+
+      // TODO: Need to actually filtering and field option
+
+      const volunteers =
+        await Volunteers.fromCommunityBusiness(knex, communityBusiness, modelQuery);
+
+      const count = has('limit', modelQuery) || has('offset', modelQuery)
+        ? await Volunteers.fromCommunityBusiness(
+            knex,
+            communityBusiness,
+            omit(['limit', 'offset'], modelQuery)
+          )
+          .then((res) => res.length)
+        : volunteers.length;
+
+      return {
+        result: await Promise.all(volunteers.map(Volunteers.serialise)),
+        meta: { total: count },
+      };
+    },
+  },
+
+  {
+    method: 'GET',
+    path: '/community-businesses/me/volunteers',
```

Comment:
`/me` has the new `user_details-parent:read` scope which allows `VOLUNTEER_ADMIN` to access volunteer user details. They should only be able to do this for their own CB.

This could be checked in a prereq, but it seemed more explicit and more inline with other routes to be defined separately.

Reply:
As per discussion, let's merge the two route definitions with a big comment above line 39 explaining that it's doing some auth stuff, and we can update issue #132 to add a pre-req to handle this case to keep handlers free of auth stuff as far as is reasonable.